src/service.c

```diff
@@ -199,6 +202,17 @@ device_compare_func (gconstpointer ga, gconstpointer gb)
         }
     }
 
+  technology = UP_DEVICE_TECHNOLOGY_UNKNOWN;
+  if (!ret && (a_technology != b_technology) &&
+      (a_technology == technology || b_technology == technology))
```

Comment:
I think you can remove this line, because if two batteries have a different technology, then they cannot be the equal.

Reply:
I'd leave this, since in case the device has no batteries with known technology I think the rest of the logic should still kick in?

Reply:
This line means that if the two batteries have different (but known) technologies, then they wouldn't be sorted according to the technology. At first sight it seems wrong, but maybe I'm missing something?

Reply:
Hmmm, it might be a bit unreadable because of the assignment to `technology` - it checks if the technologies are different and at least one of them is unknown.

If I remove the second check it would arbitrarily sort the known technologies, which I think doesn't really make sense.

Reply:
I must have misunderstood earlier. Yes, if the batteries have different but known technologies they shouldn't be sorted here, because it doesn't make sense to say that a LiPo has priority over LiFePO4.